lib/vfscore/main.c

```diff
@@ -1107,6 +1107,45 @@ UK_SYSCALL_R_DEFINE(int, getdents, int, fd, struct dirent*, dirp,
 	return (i * sizeof(struct dirent));
 }
 
+UK_TRACEPOINT(trace_vfs_getdents64, "%d %p %hu", int, struct dirent64 *, size_t);
+UK_TRACEPOINT(trace_vfs_getdents64_ret, "");
+UK_TRACEPOINT(trace_vfs_getdents64_err, "%d", int);
+
+#undef getdents64
```

Comment:
Is there any reason to keep the definition of `getdents64` as `getdents` [in `dirent.h`](https://github.com/unikraft/unikraft/blob/staging/lib/nolibc/include/dirent.h#L76)? 
If not, I would remove that and the `#undef getdents64` line.